commerce_coordinator/apps/lms/tasks.py

```diff
@@ -46,25 +46,21 @@ def on_failure(self, exc, task_id, args, kwargs, einfo):
             f"order number: {order_number}, and course title: {course_title}"
         )
 
-        # This error is returned from LMS if the course mode is unsupported
-        # https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/enrollments/views.py#L870
-        course_mode_expired_error = "course mode is expired or otherwise unavailable for course run"
-
-        if course_mode_expired_error in error_message:
-            logger.info(
-                f"Sending unsupported course mode fulfillment error email "
-                f"for the user with user ID: {edx_lms_user_id}, email: {user_email}, "
-                f"order number: {order_number}, and course title: {course_title}"
-            )
-
-            canvas_entry_properties = {
-                'order_number': order_number,
-                'product_type': 'course',  # TODO: Fetch product type from commercetools product object
-                'product_name': course_title,
-                'first_name': user_first_name,
-            }
-            # Send failure notification email
-            send_unsupported_mode_fulfillment_error_email(edx_lms_user_id, user_email, canvas_entry_properties)
+        canvas_entry_properties = {
+            'order_number': order_number,
+            'product_type': 'course',  # TODO: Fetch product type from commercetools product object
+            'product_name': course_title,
+            'first_name': user_first_name,
+        }
+
+        logger.info(
+            f"Sending unsupported course mode fulfillment error email "
+            f"for the user with user ID: {edx_lms_user_id}, email: {user_email}, "
+            f"order number: {order_number}, and course title: {course_title}"
+        )
+
+        # Send failure notification email
+        send_unsupported_mode_fulfillment_error_email(edx_lms_user_id, user_email, canvas_entry_properties)
```

Comment:
@syedsajjadkazmii Is it okay to remove the condition checks and send the email on all sorts of errors?

Reply:
you are right. I missed that we can have errors other than enrollment failure such as user matching query does not exist. 

Thanks. i will update

Reply:
updated the PR